rfcs/0138-rfc-repos.md

```diff
@@ -0,0 +1,106 @@
+---
+feature: rfc-repos
+start-date: 2022-01-10
+author: Silvan Mosberger
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: @winterqt, @lheckemann 
+shepherd-leader: @lheckemann 
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Develop and discuss RFCs in repositories instead of a pull request.
+
+# Motivation
+[motivation]: #motivation
+
+RFC discussions are currently held in single pull requests on the [rfcs](https://github.com/NixOS/rfcs/pulls) repository.
+- Due to GitHub's limitations on pull requests, this quickly becomes unmanageable.
+  The most prominent example of this is [the Flakes RFC](https://github.com/NixOS/rfcs/pull/49) with over 500 comments.
+- These PR threads easily get polluted with minor points like typos and wording improvements,
+  which are not meaningful to the RFCs contents, but distract from other more important discussions.
+- It's easiest to comment at the top-level, without threads,
+  especially when there's no relevant line of the diff to comment on for a thread.
+  This frequently causes multiple discussions to be interleaved at the top-level.
+
+# Detailed design
+[design]: #detailed-design
+
+The RFC process is changed to use repositories to develop and review RFCs instead.
+To create a new RFC, an issue on the [rfcs](https://github.com/NixOS/rfcs) repository is opened.
+It should contain the following contents describing the process:
+
+> This RFC is developed as the README.md of this repository: https://github.com/$owner/$repo
+>
+> To review the RFC
+> - Open https://github.com/$owner/$repo/blob/master/README.md?plain=1
+> - Select the lines you wish to comment on
+> - Click the "..." menu on the left
+> - Select "Reference in new issue" to open a new issue commenting on those lines
+>
+> To suggest changes, open a PR against the repository.
+>
+> All issues and pull requests must be resolved before the FCP can be initiated and completed.
+>
+> This issue may only be used for RFC meta-discussions, such as shepherd nominations, FCP periods, meeting schedules, etc.
+
+The FCP must be initiated on a specific commit of the repository.
+When the FCP passes, the repositories contents are committed to the [rfcs](https://github.com/NixOS/rfcs) repository by the [RFC steering committee](https://github.com/NixOS/rfcs#rfc-steering-committee) with a commit that closes the original issue.
+
+## Workflow to get RFC updates
+To get updates for RFC's, instead of subscribing to the PR, one has to watch the repository.
+
+## Repository transfer
+
+Once the RFC is merged, the repository has to be transferred to the NixOS organization under https://github.com/NixOS/rfc-NUMBER. This is to ensure the discussions aren't lost in the future. The repository will then be archived.
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
```

Comment:
The 2nd drawback I couldn't remember in [my previous thread](https://github.com/NixOS/rfcs/pull/138/files#r1444045425):

If an RFC has a relatively large amount of traffic, multiple discussions can start on the same logical unit discussing more or less the same thing (or even the exact same thing). While this can be considered relatively trivial to manage, it's still an overhead that I thought needs to be mentioned (which apparently already was here: https://github.com/NixOS/rfcs/pull/138#issuecomment-1377774126.

This is a consequence of the previously mentioned absence of the line-comment view.

#### Solution?

A potential way of solving this can be to have a great label architecture (much like nixpkgs). We could have "broad" labels which could be: `drawbacks`, `alternatives`, `future-work`; pretty self-explanatory but these would essentially mimic all the major headings in the RFC template. Unsure if a general rule can be applied to smaller labels, if not, they will just be a mess.